code/modules/admin/verbs/adminhelp.dm

```diff
@@ -210,6 +210,15 @@ GLOBAL_DATUM_INIT(ahelp_tickets, /datum/admin_help_tickets, new)
  * * is_bwoink - Boolean operator, TRUE if this ticket was started by an admin PM
  */
 /datum/admin_help/New(msg_raw, client/C, is_bwoink, urgent = FALSE)
+	//check message length
+	var/message_length = length(msg_raw)
+	if(message_length > MAX_MESSAGE_LEN)
+		if(C)
+			to_chat(C, span_redtext("Failed to send your adminhelp. Your message is [message_length] characters which exceeds the maximum length of [MAX_MESSAGE_LEN], please reduce the length and try again. You attempted to send the following message:\n[sanitize(msg_raw)]"))
+		message_admins("[C ? C.ckey : "An unknown client"] attempted to send an adminhelp that was too long.")
+		qdel(src)
+		return
```

Comment:
Would be nice if this was done on the ui act of the adminhelp so the person writing the ahelp wouldn't have their entire message deleted, idk if the html one does similar.

Reply:
I thought about putting the limit in the TGUI window, but there's also the fallback window it uses if TGUI fails for some reason, and wanted it to work on both.

Reply:
@san7890 you did this type of change for the OOC messages, thoughts on this one?

Reply:
thoughts on what? feeding back the message is just good to do in TGUI or HTML format